tap_salesforce/salesforce/__init__.py

```diff
@@ -381,13 +387,25 @@ def _get_selected_properties(self, catalog_entry):
                                             self.select_fields_by_default)]
 
 
-    def get_start_date(self, state, catalog_entry):
+    def get_start_date(self, state, catalog_entry, with_lookback=False):
+        """
+            if 'with_lookback' is False, then return start date or state file date
+            else subtract lookback window from the state file date and return
+                if no state file bookmark is found, do not subtract lookback window from the start date
+        """
```

Comment:
Update this comment to match the new implementation

Reply:
Updated the comment as per the suggested implementation.